apps/test/unit/templates/sectionProgressV2/IconKeyTest.jsx

```diff
@@ -1,41 +1,60 @@
 import React from 'react';
 import {fireEvent, render, screen} from '@testing-library/react';
 import {expect} from '../../../util/reconfiguredChai';
+import * as utils from '@cdo/apps/utils';
+import sinon from 'sinon';
 import IconKey from '@cdo/apps/templates/sectionProgressV2/IconKey';
 
 describe('IconKey Component', () => {
   it('renders the open state initially', () => {
+    sinon.stub(utils, 'tryGetLocalStorage').returns('true');
     render(<IconKey isViewingValidatedLevel={false} expandedLessonIds={[]} />);
     expect(screen.getByLabelText('Icon key')).be.visible;
     expect(screen.queryByText('More Details')).be.visible;
     expect(screen.getByText('Assignment Completion States')).to.exist;
     expect(screen.getByText('Teacher Actions')).to.exist;
     expect(screen.queryByText('Level Types')).to.be.null;
+    utils.tryGetLocalStorage.restore();
+  });
+
+  it('renders the closed state when localStorage indicates the key is closed', () => {
+    sinon.stub(utils, 'tryGetLocalStorage').returns('false');
+    render(<IconKey isViewingValidatedLevel={false} expandedLessonIds={[]} />);
+    expect(screen.queryByText('Assignment Completion States')).to.be.null;
+    expect(screen.queryByText('Teacher Actions')).to.be.null;
+    expect(screen.queryByText('Level Types')).to.be.null;
+    utils.tryGetLocalStorage.restore();
   });
 
   it('expands collapses on click', () => {
+    sinon.stub(utils, 'tryGetLocalStorage').returns('true');
```

Comment:
Optional nit: You can move `utils.tryGetLocalStorage.restore()` to an `afterEach()` method.

And you could also try `sinon.stub(utils, 'tryGetLocalStorage').returns('true');` in a beforeEach() and restub with false where necessary. Both optional

Reply:
Yeah, I did consider both of these... good to know it is optional in your mind too.  It seemed more annoying to restub and we didn't need to do `utils.tryGetLocalStorage.restore()` after every test so I decided to list it out... I think I will keep it as is for now.  I was torn on it myself too.